app/src/main/java/fr/neamar/kiss/dataprovider/ContactsProvider.java

```diff
@@ -41,6 +43,14 @@ public void reload() {
             } else if ((matchPositionStart = contactNameNormalized.indexOf(queryWithSpace)) > -1) {
                 relevance = 40;
                 matchPositionEnd = matchPositionStart + queryWithSpace.length();
+            } else if (contact.nickname.contains(query)) {
+                alias = true;
+                contact.displayName = contact.name
```

Comment:
(see comment below first)

Same thing, just a little concerned with performance.

If I have 10k contacts, this is adding a lot of tests -- string comparison, true, but still a lot of probably useless tests (I don't expect anyone to have more than 10% of his contacts with nicknames).

Maybe we can instead store a second list on the ContactProviders (`pojosWithAlias`) and iterate over this list directly,  therefore skipping most of the empty tests?

I may be wrong. This is clearly open for discussion!

(@saveman71  / @nmitsou: any thoughts?)